### PR TITLE
uploaded watering can updates for fluffymachines.

### PR DESCRIPTION
**Pull request by Rene98**

### CODE REVIEW

**Comment by NCBPFluffyBear** [comment]
This should be coded to work with this plugin, not Fluffymachines

File: src/main/java/io/ncbpfluffybear/supserv/items/WateringCan.java [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+          0.3);
+  public final ItemSetting<Double> cropSuccessChance = new ItemSetting<>(this, "crop-success-chance", 0.3);
+  public final ItemSetting<Double> treeSuccessChance = new ItemSetting<>(this, "tree-success-chance", 0.3);
+  public final ItemSetting<Double> exoticGardenSuccessChance = new ItemSetting<>(this, "exotic-garden-success-chance", 0.3);
+
+  private static final int USE_INDEX = 7;
+  private static final int MAX_SUGAR_GROW_HEIGHT = 5;
+  private static final NamespacedKey usageKey = new NamespacedKey(SupServPlugin.getInstance(), "watering_can_usage");
+
+  private final canType canType;
```
Comment (NCBPFluffyBear) [inline src/main/java/io/ncbpfluffybear/supserv/items/WateringCan.java:49]: Move private variables under class constants

Comment (NCBPFluffyBear) [reply src/main/java/io/ncbpfluffybear/supserv/items/WateringCan.java:49]: The can type is also never needed, just the size. You can instead create a private variable for the integer size and then assign it in the constructor.

File: src/main/java/io/ncbpfluffybear/supserv/items/WateringCan.java [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+  private static final int GOLD_SIZE = 250;
+  private static final int DIAMOND_SIZE = 400;
+  private static final int EMERALD_SIZE = 650;
+  private static final int NETHERITE_SIZE = 1000;
+
+  public WateringCan(Category category, SlimefunItemStack item, RecipeType recipeType,
+                     ItemStack[] recipe, canType size) {
+    super(category, item, recipeType, recipe);
+    this.canType = size;
+    addItemSetting(maxUses);
```
Comment (NCBPFluffyBear) [inline src/main/java/io/ncbpfluffybear/supserv/items/WateringCan.java]: You have this ItemSetting, but it seems that it doesn't actually affect the size

**Comment by NCBPFluffyBear** [comment]
One more thing: Can you add some kind of prefix or suffix to the IDs? Right now, there is an item registration conflict because the same Watering can ID is used for SupServ and FM

**Comment by NCBPFluffyBear** [comment]
There are still some conflicts, please fix

**Comment by NCBPFluffyBear** [comment]
> One more thing: Can you add some kind of prefix or suffix to the IDs? Right now, there is an item registration conflict because the same Watering can ID is used for SupServ and FM

This is still an issue, once the ID conflict occurs, FM is disabled. Discussed this issue with Skizz and he decided it would be best if there was an additional recipe using the FM watering can to craft the tier 2 SS can

File: src/main/java/io/ncbpfluffybear/supserv/items/WateringCan.java (hunk 2 above)
Comment (Rene98) [reply src/main/java/io/ncbpfluffybear/supserv/items/WateringCan.java]: Your totally right, this is not needed anymore since enum of watercanTypes

**Comment by NCBPFluffyBear** [comment]
Do you think we should have a label on the second recipe saying it uses the FM watering can to avoid confusion?
Or, we could have skizzles just hide the recipe from the guide

**Comment by NCBPFluffyBear** [comment]
Ok skizz said that he would hide it. In that case, could you change the lore so it doesnt look like a watering can? like put some kind of tag saying it shouldnt be obtainable

**Comment by NCBPFluffyBear** [comment]
Resolve conflicts

**Comment by Skizzles** [comment]
There is also an error with this area:
https://screenshots.skizzles.me/wAnA3/peBIsOrU75.png

This error stops the upgrading still as this recipe does not register.

**Pull request merged**